src/mqt/predictor/ml/Predictor.py

```diff
@@ -32,143 +33,169 @@ def __init__(self, verbose: int = 0) -> None:
-                                            device_name,
-                                            opt_level,
-                                            False,
-                                            False,
-                                            target_path,
-                                            target_filename,
-                                        ],
-                                        timeout,
-                                    )
-                                    results.append(tmp)
-                                    if not tmp:
-                                        continue
-                        elif "tket" in compiler:
-                            for lineplacement in settings["lineplacement"]:
-                                target_filename = filename.split(".qasm")[0] + "_" + str(comp_path_id)
-                                comp_path_id += 1
-                                if max_qubits >= qc.num_qubits:
-                                    tmp = utils.timeout_watcher(
-                                        tket_helper.get_mapped_level,
-                                        [
-                                            qc,
-                                            gate_set_name,
-                                            qc.num_qubits,
-                                            device_name,
-                                            lineplacement,
-                                            False,
-                                            False,
-                                            target_path,
-                                            target_filename,
-                                        ],
-                                        timeout,
-                                    )
-                                    results.append(tmp)
-                                    if not tmp:
-                                        continue
-
-            if all(x is False for x in results):
-                logger.debug("No compilation succeeded for this quantum circuit: " + filename)
-                return False
-            return True
+            qc = QuantumCircuit.from_qasm_file(Path(source_path) / filename)
+            if filename.suffix != ".qasm":
+                return
+
+            for i, dev in enumerate(rl.helper.get_devices()):
+                target_filename = str(filename).split("/")[-1].split(".qasm")[0] + "_" + figure_of_merit + "_" + str(i)
+                if (
+                    not (Path(target_path) / (target_filename + ".qasm")).exists()
+                    and qc.num_qubits <= dev["max_qubits"]
+                ):
+                    try:
+                        res = utils.timeout_watcher(rl.qcompile, [qc, figure_of_merit, dev["name"]], timeout)
+                        if res:
+                            compiled_qc = res[0]
+                            compiled_qc.qasm(filename=Path(target_path) / (target_filename + ".qasm"))
```

Comment:
True. I remember that we in "the early days" of MQT Bench used that format but changed to qasm. I cannot recall the exact reasons for that, but maybe since TKET could only produce QPY via a workaround of going through Qiskit.QuantumCircuit again.